### ISSUE TITLE
Flatpak pinentry not working on Alpine

### ISSUE THREAD

**Issue by K0-RR** [quexten/goldwarden#66]
![image](https://github.com/quexten/goldwarden/assets/43277609/84a4daee-2802-4fac-bf0d-fccd3b74c6f4)

**Comment by quexten** [comment]
Hi, thanks for the issue report. I guess the current initial view is not self-explaining. You have to set up a pin first. After that, the login button will be enabled.

**Comment by K0-RR** [comment]
Ooh. Just tried but it doesn't ask me for the pin. Not sure if there should be another pop-up the enter it or something else. Here is the log obtained via the `flatpak run` command:
```
/app/bin/monitors/dbus_autofill_monitor.py:4: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '4.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk
IS daemon running False
running daemon
Failed err None
Flatpak Config directory: /home/k0rr/.var/app/com.quexten.Goldwarden/config/goldwarden.json

[INF] [00:47] [Goldwarden > Keyring] >>> Creating new memguard keyring

[WRN] [00:47] [Goldwarden > Agent] >>> Config is not locked. SET A PIN!!

[INF] [00:47] [Goldwarden > Agent] >>> Agent listening on /home/k0rr/.var/app/com.quexten.Goldwarden/data/goldwarden.sock...

[INF] [00:47] [Goldwarden > SSH] >>> SSH Agent listening on /home/k0rr/.var/app/com.quexten.Goldwarden/data/ssh-auth-sock

Blocking, press ctrl+c to continue...

[WRN] [00:47] [Goldwarden > Agent] >>> Could not monitor idle: org.freedesktop.DBus.Error.ServiceUnknown

autostart enabled..!?
(':1.1313', '/org/freedesktop/portal/desktop/request/1_1347/com/quexten/Goldwarden/17705351', 'org.freedesktop.portal.Request', 'Response', GLib.Variant('(ua{sv})', (0, {'background': <true>, 'autostart': <true>})), None)
/app/bin/settings.py:34: DeprecationWarning: Adw.ActionRow.set_icon_name is deprecated
  self.ssh_row.set_icon_name("emblem-default")
[INF] [00:48] [Goldwarden > Systemauth] >>> Checking permission for goldwarden with session type com.quexten.goldwarden.accessvault

[WRN] [00:48] [Goldwarden > Biometrics] >>> Only pkexec permissions found, consider installing polkit policies

[INF] [00:48] [Goldwarden > Biometrics] >>> Checking biometrics for com.quexten.goldwarden.accessvault

[ERR] [00:48] [Goldwarden > Biometrics] >>> Failed to create polkit authority: Action com.quexten.goldwarden.accessvault is not registered

[INF] [00:48] [Goldwarden > Biometrics] >>> Falling back to pkexec permissions

[INF] [00:48] [Goldwarden > Biometrics] >>> Biometrics result: true

[INF] [00:48] [Goldwarden > Systemauth] >>> Permission granted, creating session

[INF] [00:48] [Goldwarden > Pinentry] >>> Asking for pin |Pin Change|Enter your desired pin|

```

It I click "Set Pin" again it prints the last part again:
```
[INF] [00:48] [Goldwarden > Pinentry] >>> Asking for pin |Pin Change|Enter your desired pin|

[INF] [00:48] [Goldwarden > Systemauth] >>> Checking permission for goldwarden with session type com.quexten.goldwarden.accessvault

[INF] [00:48] [Goldwarden > Systemauth] >>> Permission granted from cached session
```

**Comment by K0-RR** [comment]
Done, still not working correctly. There may be something missing with flatpaks on my side (portals maybe? I'm on Alpine Linux), as I noticed any flatpak app can't open a file explorer window when clicking a button that should do it.

```
core% flatpak run com.quexten.Goldwarden 
/app/bin/monitors/dbus_autofill_monitor.py:4: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '4.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk
IS daemon running False
running daemon
Failed err None
Flatpak Config directory: /home/k0rr/.var/app/com.quexten.Goldwarden/config/goldwarden.json

[INF] [11:01] [Goldwarden > Keyring] >>> Creating new memguard keyring

[WRN] [11:01] [Goldwarden > Agent] >>> Could not read config: open /home/k0rr/.var/app/com.quexten.Goldwarden/config/goldwarden.json: no such file or directory

[WRN] [11:01] [Goldwarden > Agent] >>> Config is not locked. SET A PIN!!

[INF] [11:01] [Goldwarden > Agent] >>> Agent listening on /home/k0rr/.var/app/com.quexten.Goldwarden/data/goldwarden.sock...

Blocking, press ctrl+c to continue...

[INF] [11:01] [Goldwarden > SSH] >>> SSH Agent listening on /home/k0rr/.var/app/com.quexten.Goldwarden/data/ssh-auth-sock

[WRN] [11:01] [Goldwarden > Agent] >>> Could not monitor idle: org.freedesktop.DBus.Error.ServiceUnknown

autostart enabled..!?
(':1.1737', '/org/freedesktop/portal/desktop/request/1_1811/com/quexten/Goldwarden/10453872', 'org.freedesktop.portal.Request', 'Response', GLib.Variant('(ua{sv})', (0, {'background': <true>, 'autostart': <true>})), None)
/app/bin/settings.py:34: DeprecationWarning: Adw.ActionRow.set_icon_name is deprecated
  self.ssh_row.set_icon_name("emblem-default")
[INF] [11:01] [Goldwarden > Pinentry] >>> Asking for pin |Pin Change|Enter your desired pin|

[INF] [11:02] [Goldwarden > Pinentry] >>> Asking for pin |Pin Change|Enter your desired pin|

[INF] [11:02] [Goldwarden > Pinentry] >>> Asking for pin |Pin Change|Enter your desired pin|

[INF] [11:02] [Goldwarden > Pinentry] >>> Asking for pin |Pin Change|Enter your desired pin|

[INF] [11:02] [Goldwarden > Pinentry] >>> Asking for pin |Pin Change|Enter your desired pin|
```

**Comment by quexten** [comment]
Ah, could very well be as alpine is usually quite stripped down. Though portals should only be required for auto-type. In this case it's either polkit, or pinentry that's not working.
What desktop environment are you on?

**Comment by K0-RR** [comment]
KDE. I just installed Gnome to test if the issue will be present there and after starting Goldwarden I was asked for Pin. I cancelled that window and switched back to KDE. After trying again on it, it's now working here too so I think some package was missing but got installed with the Gnome.

**Comment by quexten** [comment]
Probably fixed in #100, still needs testing.

**Comment by quexten** [comment]
Should be fixed in v0.2.12. If not please re-open.
